### PR TITLE
Updates to the README for release.

### PR DESCRIPTION
**Pull request by crspeller**
Combination of @it33's earlier PRs. @it33 Please review. 

Some additional changes as well:
- Changing docker instructions to use implicit :latest instead of :helium. :latest will point at the latest release. 
- Head will be :dev (not implemented yet)
- Some more links

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -128,6 +131,8 @@ Contributing
 
 To contribute to this open source project please review the Mattermost Contribution Guidelines at http://www.mattermost.org/contribute-to-mattermost/.
 
+To seup your machine for development of mattermost see: [Developer Machine Setup](scripts/README_DEV.md)
```
Comment (hmhealey) [inline README.md]: Typo here

**Comment by hmhealey** [comment]
Looks good other than that small typo

**Comment by crspeller** [comment]
Typo fixed waiting for @it33

**Comment by it33** [comment]
Proposing minor adjustments: https://github.com/mattermost/platform/pull/334

**Comment by crspeller** [comment]
@it33's changes have been integrated into this pull request. 

@it33 @hmhealey Give your +1 and I will merge this.

**Comment by it33** [comment]
Looks good!

**Pull request merged**